### PR TITLE
1540 on hover the column headers display the filter keyword that enables filtering by that column

### PR DESCRIPTION
**Pull request by sergioloporto**
As agreed in the Thursday's call, tooltip removed.

I made the column header text clickable, and after click the corresponding filter is added.
Added a mechanism to avoid adding duplicated filters from clicking the column header.
Added a title to the header, so if the user puts the cursor on the column header text for 1-2 seconds, appears the text title near the cursor.

### CODE REVIEW

**Review by maciekz1996 — CHANGES_REQUESTED** [review]
Adding filters to Search Bar by clicking on column headers works great 💯 

There are some issues that I've found. Most of them are marked in the code and explained in the comments. 

There also seems to be a problem with displaying search errors under `SearchBar` component which is also why tests are failing. This should be looked more deeply into.

**Steps to reproduce**
* Click on `country` column header
* In Search Bar enter blank space and wait
* No error message is being shown and `LineListTable` is not being updated (but the request is being made) 
* Then after typing for example "china" in Search Bar the error message shows up, but it shouldn't be there

File: verification/curator-service/ui/amplify/backend/function/curatoruic254d982CreateAuthChallenge/src/boilerplate-create-challenge.js [hunk 1]
```diff
@@ -11,23 +11,25 @@ exports.handler = async (event, context) => {
         // Generate a new secret login code and mail it to the user
```
[the remark this answers is not in the document]
Comment (sergioloporto) [reply verification/curator-service/ui/amplify/backend/function/curatoruic254d982CreateAuthChallenge/src/boilerplate-create-challenge.js]: file removed from commit

File: verification/curator-service/ui/src/components/LinelistTable.tsx [hunk 2]
```diff
@@ -814,7 +853,20 @@ class LinelistTable extends React.Component<Props, LinelistTableState> {
                             },
                         },
                         {
-                            title: 'Case ID',
+                            title: (
```
[the remark this answers is not in the document]
Comment (sergioloporto) [reply verification/curator-service/ui/src/components/LinelistTable.tsx:876]: done

File: verification/curator-service/ui/src/components/ColumnHeaderTooltip.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { AppTooltip } from '../components/common-form-fields/AppTooltip';
```
[the remark this answers is not in the document]
Comment (sergioloporto) [reply verification/curator-service/ui/src/components/ColumnHeaderTooltip.tsx:1]: removed

**Review by sergioloporto — COMMENTED** [review]
> There are some issues that I've found. Most of them are marked in the code and explained in the comments.
> 
> There also seems to be a problem with displaying search errors under `SearchBar` component which is also why tests are failing. This should be looked more deeply into.
> 
> **Steps to reproduce**
> 
> * Click on `country` column header
> * In Search Bar enter blank space and wait
> * No error message is being shown and `LineListTable` is not being updated (but the request is being made)
> * Then after typing for example "china" in Search Bar the error message shows up, but it shouldn't be there

I tested the filtering that was before applying filtering from the columns.
This seems to be a problem that is in the SearchBar, and is unrelated to the filtering from the columns'headers.

This can be reproduced this way:
* Click on `country` from the FILTER menu
* In Search Bar enter blank space and wait
* No error message is being shown and `LineListTable` is not being updated (but the request is being made)
* Then after typing for example "china" in Search Bar the error message shows up, but it shouldn't be there

so this seems to confirm that this is a problem related to the search component logic, not to the filtering from the columns headers

**Review by maciekz1996 — APPROVED** [review]

**Pull request merged**